tests/test_dependency_rendering_e2e.py

```diff
@@ -539,6 +539,58 @@ async def test_fragment_htmx(self):
 
         await page.close()
 
+    # Fragment where the page wasn't rendered with the "document" strategy
+    @with_playwright
+    async def test_fragment_without_document(self):
```

Comment:
This tests that we can fetch and insert the fragments and they fetch their own JS/CSS